UndertaleModTool/Editors/UndertaleRoomEditor.xaml.cs

```diff
@@ -1221,22 +1237,75 @@ private void AddLayer<T>(LayerType type, string name) where T : Layer.LayerData,
                 }
             }
 
+            int layerDepth = 0;
+            if (room.Layers.Count > 0)
+            {
+                layerDepth = room.Layers.Select(l => l.LayerDepth).Max();
+                if (layerDepth + 100 > int.MaxValue)
+                {
+                    if (layerDepth + 1 > int.MaxValue)
+                    {
+                        layerDepth -= 1;
+                        MainWindow.ShowWarning("Warning - a maximum layer depth is reached.\nYou probably should change the depth of the new layer.");
+                    }
+                    else
+                        layerDepth += 1;
+                }
+                else
+                    layerDepth += 100;
+            }
+
+            string baseName = null;
+            int nameNum = 0;
+            while (room.Layers.Any(l => l.LayerName.Content == name))
+            {
+                if (baseName is null)
+                {
+                    // "name123" => "123"
+                    Match numMatch = endNumberEx.Match(name);
+
+                    if (numMatch.Success)
+                    {
+                        if (baseName is null)
+                        {
+                            baseName = name[..^numMatch.Length];
+                            nameNum = Int32.Parse(numMatch.Groups[0].Value) + 1;
+                        }
+                        else
+                            nameNum++;
+
+                        // "name9" => "name10" 
+                        name = baseName + nameNum;
+                    }
+                    else
+                    {
+                        // "name" => "name1"
+                        name += 1;
+                    }
+                }
+                else
+                    name = baseName + (++nameNum);
```

Comment:
Would prefer if this whole thing were a tad better commented.

Reply:
But is it necessary to insert comments between `if-else`, so they make a gap between them?

Reply:
E.g.
```
}
// if base name is already extracted, increment "nameNum" and append it to the base name
else
```

Reply:
Also, why you've extracted `nameNum++` to a separate line?
Why not using the possibility to simplify that to `(++nameNum)` ?

Reply:
> But is it necessary to insert comments between if-else, so they make a gap between them?

Because that actually makes it clearer from the outside what the individual chunks do.

> Why not using the possibility to simplify that to (++nameNum) ?

because it's harder to read.

Reply:
And what about this?
```c#
}
else // if base name is already extracted, increment "nameNum" and append it to the base name
{
```

Reply:
Hmm... then fine.
But the second null check for `baseName` is for extracting the name and number only once, so it shouldn't be deleted.

Reply:
It is an impossible scenario.
There is no way that basename is null in the first null check, but not null in the second